tests/integration/deleted_with/go/fails_on_delete.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2016-2023, Pulumi Corporation.  All rights reserved.
```

Comment:
It might be worth considering moving `testprovider` to its own repo and publishing it. Then we could use it from this repo and others (like `pulumi-dotnet`, `pulumi-java`, etc.), and we wouldn't need to have multiple copies of these hand-written files for calling the resources in the provider in each test that uses them.

Reply:
I think pretty much the entire integration test set should be callable from each language repo, not sure if that means testprovider needs to be moved out to it's own repo as well or not though. And as you'll have seen in dotnet I re-wrote the testprovider to C# in that repo, might make sense to do that for other languages as well.